### PR TITLE
fix: clean up TS errors

### PR DESCRIPTION
**Pull request by anthony9187**
## Description

This PR cleans up a lot of TS errors throughout the repo in preparation of switching our build step to use the TypeScript compiler. There's a bit of leg work that needs to be done(or maybe just some configuration) before that can happen, so I'd like to get these changes merged, and then capture that switch in a subsequent PR

note: in FPSCounter, I've added `.toString()` calls to all the numbers since `Text` wants strings, not numbers. Is this how we'd prefer to do that, or do we want to loosen the children type of Text?

## Changes

- improves type safety of various components
- adds ts-expect-error lines where we're waiting on updates from outside libraries
- references the l3-ui-theme-base for the theme type for development

## Testing

ensure we're no longer receiving type errors inside the packages/solid workspace, ensure all components render and function as expected

### CODE REVIEW

**Comment by chiefcll** [comment]
I would read all the files... but... I have a life. I'll assume you know what you're doing on this one...

**Comment by chiefcll** [comment]
@slhay28 can you just test the PR by opening the branch and running storybook

File: src/packages/solid/components/Tile/Tile.tsx [hunk 1]
```diff
@@ -223,8 +224,9 @@ const Tile: Component<TileProps> = (props: TileProps) => {
           {props.bottom}
         </View>
       </Show>
-
-      <Show when={props.progressBar?.progress ? props.progressBar.progress > 0 : 0}>
+      {/* TODO why was this a ternary? */}
+      <Show when={props.progressBar.progress > 0}>
```
Comment (slhay28) [inline src/packages/solid/components/Tile/Tile.tsx:228]: progressBar can be undefineid- need `props.progressBar?.progress`

File: src/packages/solid/components/Key/Key.tsx [hunk 2]
```diff
@@ -47,14 +47,13 @@ export interface KeyProps extends UIComponentProps {
    * If true, pressing the key will trigger the $toggleKeyboard event. If false, the key will trigger the $onSoftKey event.
    */
   toggle?: boolean;
-
-  keySignal: Partial<KeyStyles>;
```
Comment (slhay28) [inline src/packages/solid/components/Key/Key.tsx:51]: this prop is called in `SimpleKeyboard` line 27

File: src/packages/solid/components/Tile/Tile.tsx (hunk 1 above)
Comment (anthony9187) [reply src/packages/solid/components/Tile/Tile.tsx:228]: i think we're still good to pull it out of the ternary though, right?

**Pull request merged**